test/test_autograd.py

```diff
@@ -5450,6 +5450,56 @@ def test_advanced_indexing_backwards_large(self, device):
         a.sum().backward()
         self.assertEqual(x.grad, torch.ones(n, 1, device=device))
 
+    @onlyCUDA
+    def test_reentrant_parent_error_on_cpu(self, device):
+        before = CudaMemoryLeakCheck.get_cuda_memory_usage()
+        t1 = torch.rand([3, 3], requires_grad=True)
+        t2 = torch.rand([3, 3], device=device, requires_grad=True)
+        t3 = torch.rand([3, 3], device=device, requires_grad=True)
+
+        # Parent graph cpu graph.
+        t4 = t1 * t1
+        t5 = TestAutograd.BackwardError.apply(t4)
+
+        # Child gpu graph (much longer than parent graph).
+        computations = []
+        computations.append(t2 * t2)
+        for i in range(10):
+            computations.append(computations[-1] * t2)
+        reentrant_root = computations[-1].sum()
+
+        class ReentrantFunc(Function):
+            @staticmethod
+            def forward(ctx, inp):
+                return inp
```

Comment:
nit: in general avoid empty forwards in Function. It is a special case that can lead to weird behavior when mixed with inplace. I usually do `return inp.clone()`.

Reply:
Thanks for the suggestion, although which inplace operation are you referring to here?

Reply:
There is not here, it's a nit. So it is fine to do it here. It is more of a general guideline when writing custom Functions.